docker_test.go

```diff
@@ -122,3 +122,34 @@ func TestExternalVolume(t *testing.T) {
 	assert.Equal(t, ev2.Name, "")
 	assert.Equal(t, ev2.Provider, "")
 }
+
+func TestDockerPersistentVolume(t *testing.T) {
+	docker := NewDockerApplication()
+
+	pVol := &PersistentVolume{Size: 2048}
+	pVol.SetType(PersistentVolumeTypeRoot)
+	pVol.AddConstraint("bat")
+
+	assert.Equal(t, 1, len(*pVol.Constraints))
+	assert.Equal(t, []string{"bat"}, (*pVol.Constraints)[0])
+	pVol.EmptyConstraints()
+	assert.NotNil(t, pVol.Constraints)
+	assert.Equal(t, 0, len(*pVol.Constraints))
+
+	container := docker.Container.Volume("/host", "/container", "RW")
+	assert.Equal(t, 1, len(*docker.Container.Volumes))
+
+	vol := (*container.Volumes)[0]
+	pVol = vol.SetPersistentVolume(256)
+	pVol.SetType(PersistentVolumeTypeMount)
+	pVol.SetMaxSize(128)
+
+	assert.Equal(t, "/host", vol.HostPath)
```

Comment:
Please do not validate the Volume-only parameters (i.e., drop this line and the next two). This should be the responsibility of the Volume-specific tests.